### PR TITLE
fix(replaceAll): add TC39 proposal-compliant String.replaceAll ponyfill to use in RA.replaceAll 

### PR DESCRIPTION
**Pull request by js636f**

### CODE REVIEW

**Comment by char0n** [comment]
I'm rerunning failed tests for node 9. Doing CR now. Thank you!

File: src/replaceAll.js [hunk 1]
```diff
@@ -1,34 +1,33 @@
...
- * @throws {Error} When `searchValue` is RegExp
  * @see {@link http://ramdajs.com/docs/#replace|R.replace}, {@link https://github.com/tc39/proposal-string-replaceall|TC39 proposal}
  * @example
  *
  * RA.replaceAll('ac', 'ef', 'ac ab ac ab'); //=> 'ef ab ef ab'
+ * RA.replaceAll('', '_', 'xxx'); //=> '_x_x_x_'
+ * RA.replaceAll(/x/g, 'v', 'xxx'); //=> 'vvv'
+ * RA.replaceAll(/x/, 'v', 'xxx'); //=> TypeError
  */
 const replaceAll = curryN(3, (searchValue, replaceValue, str) => {
```
Comment (char0n) [inline src/replaceAll.js]: We have a convention that standardizes the way we use polyfills. Take a look at [isInteger](https://github.com/char0n/ramda-adjunct/blob/6e9d1a91a26e45ac923ea3a90cab8274e9d6222d/src/isInteger.js#L27) and how to create [tests](https://github.com/char0n/ramda-adjunct/blob/6e9d1a91a26e45ac923ea3a90cab8274e9d6222d/test/isInteger.js#L8-L7) for it.

File: test/replaceAll.js [hunk 2]
```diff
@@ -47,4 +42,65 @@ describe('replaceAll', function() {
     assert.strictEqual(RA.replaceAll('a', 'c')('aba'), 'cbc');
     assert.strictEqual(RA.replaceAll('a')('c')('aba'), 'cbc');
   });
+  context('given searchValue is a non-global RegExp', function() {
```
Comment (char0n) [inline test/replaceAll.js]: Insert empty line here pls.

File: src/replaceAll.js (hunk 1 above)
Comment (js636f) [reply src/replaceAll.js]: @char0n, I hopeI get it right (I made a wrapper for standard .replaceAll to save RA.replaceAll API and curried it). Also I changed the polyfill API to match existing RA.ReplaceAll.

File: src/replaceAll.js [hunk 3]
```diff
@@ -10,25 +13,26 @@ import isRegExp from './isRegExp';
...
-const replaceAll = curryN(3, (searchValue, replaceValue, str) => {
-  if (isRegExp(searchValue)) {
-    throw new Error('searchValue must be a String, not a RegExp.');
-  }
 
-  return pipe(
-    split(String(searchValue)),
-    join(String(replaceValue))
-  )(str);
+const standardReplaceAll = curryN(3, (searchValue, replaceValue, str) => {
```
Comment (char0n) [inline src/replaceAll.js]: You can use invoker in functional programming to acheive the same result with less code. Here is an example what we already have in codebase: https://github.com/char0n/ramda-adjunct/blob/f1c5dfaac3b38a5599a6bb71a570205b025c7112/src/repeatStr.js#L8
```js
const replaceAllInvoker = invoker(2, 'replaceAll');
```
Here is documentation for invoker: https://ramdajs.com/docs/#invoker

File: test/replaceAll.js [hunk 4]
```diff
@@ -47,4 +42,66 @@ describe('replaceAll', function() {
...
   });
+
+  context('given searchValue is a non-global RegExp', function() {
+    specify('should throw Error', function() {
+      assert.throws(() => RA.replaceAll(/a/, 'c', 'abc'));
+    });
+  });
+});
+
+describe('replaceAllPolyfill', function() {
```
Comment (char0n) [inline test/replaceAll.js]: We're missing currying test for polyfill which should be the same as this one: https://github.com/char0n/ramda-adjunct/blob/f4cca77cde95afdeafbf6c5604731a19bc99a2f7/test/replaceAll.js#L39

File: test/replaceAll.js [hunk 5]
```diff
@@ -47,4 +42,66 @@ describe('replaceAll', function() {
     assert.strictEqual(RA.replaceAll('a', 'c')('aba'), 'cbc');
     assert.strictEqual(RA.replaceAll('a')('c')('aba'), 'cbc');
   });
+
+  context('given searchValue is a non-global RegExp', function() {
+    specify('should throw Error', function() {
+      assert.throws(() => RA.replaceAll(/a/, 'c', 'abc'));
```
Comment (char0n) [inline test/replaceAll.js]: Pls assert the throw error type in every test you wrote:
```js
assert.throws(() => RA.replaceAll(/a/, 'c', 'abc'), TypeError);

File: src/replaceAll.js [hunk 6]
```diff
@@ -10,25 +15,22 @@ import isRegExp from './isRegExp';
  * @since {@link https://char0n.github.io/ramda-adjunct/2.17.0|v2.17.0}
```
Comment (char0n) [inline src/replaceAll.js]: I've changed version back to 2.17.0 as this is the version that the function was introduced to codebase.

**Comment by js636f** [comment]
@char0n, I'm happy to help! Thank you for your support!

**Pull request merged**